Dockerfile

```diff
@@ -18,56 +18,37 @@
 FROM node:18-bookworm as build
 RUN node --version && npm --version
 
-# Prepare the folder to enable non-root, otherwise npm will refuse to run the postinstall
-RUN mkdir /vault
-RUN chown node:node /vault
-USER node
-
 # Can be a tag, release, but prefer a commit hash because it's not changeable
 # https://github.com/bitwarden/clients/commit/${VAULT_VERSION}
 #
 # Using https://github.com/bitwarden/clients/releases/tag/web-v2024.5.1
 ARG VAULT_VERSION=9823f69c9d17e2d94de1cc005e01202dd95f0647
+ENV VAULT_VERSION=$VAULT_VERSION
+ENV VAULT_FOLDER=bw_clients
+ENV CHECKOUT_TAGS=false
 
-WORKDIR /vault
-RUN git -c init.defaultBranch=main init && \
-    git remote add origin https://github.com/bitwarden/clients.git && \
-    git fetch --depth 1 origin "${VAULT_VERSION}" && \
-    git -c advice.detachedHead=false checkout FETCH_HEAD
-
-COPY --chown=node:node patches /patches
-COPY --chown=node:node resources /resources
-COPY --chown=node:node scripts/apply_patches.sh /apply_patches.sh
-
-RUN bash /apply_patches.sh
-
-# Build
-RUN npm ci
+RUN mkdir /bw_web_builds
+WORKDIR /bw_web_builds
 
-# Switch to the web apps folder
-WORKDIR /vault/apps/web
+COPY patches ./patches
+COPY resources ./resources
+COPY scripts ./scripts
 
-RUN npm run dist:oss:selfhost
+RUN ./scripts/checkout_web_vault.sh
+RUN	./scripts/patch_web_vault.sh
+RUN	./scripts/build_web_vault.sh
+RUN	mv $VAULT_FOLDER/apps/web/build ./web-vault
```

Comment:
The spacing here doesn't look right.

Reply:
which spacing?
Looks fine too me.

The only remark i might have is that it might be better to use `mv "${VAULT_FOLDER}/apps/web/build" ./web-vault` though.

Reply:
Just compare it with the line above the part I selected:
`RUN ./scripts/checkout_web_vault.sh`
vs
```
RUN	./scripts/patch_web_vault.sh
RUN	./scripts/build_web_vault.sh
RUN	mv $VAULT_FOLDER/apps/web/build ./web-vault
```
I think he might have use tabs instead of spaces for these 3 lines ^^